gateway/src/shard/processor/impl.rs

```diff
@@ -78,23 +84,37 @@ impl ShardProcessor {
         });
 
         let session = Arc::new(Session::new(tx));
+        if resumable {
+            session
+                .id
+                .lock()
+                .await
+                .replace(config.session_id.clone().unwrap());
+            session
+                .seq
+                .store(config.sequence.unwrap(), Ordering::Relaxed)
+        }
 
         let (wtx, wrx) = watch_channel(Arc::clone(&session));
 
-        Ok((
-            Self {
-                config,
-                listeners: Listeners::default(),
-                properties,
-                rx,
-                session,
-                inflater: Inflater::new(shard_id),
-                url,
-                resume: None,
-                wtx,
-            },
-            wrx,
-        ))
+        let mut processor = Self {
+            config,
+            listeners: Listeners::default(),
+            properties,
+            rx,
+            session,
+            inflater: Inflater::new(shard_id),
+            url,
+            resume: None,
+            wtx,
+        };
+
+        if resumable {
+            debug!("Shard {:?} resuming", shard_id);
+            processor.resume().await;
```

Comment:
This returns a result, which should probably either be handled or ignored with a `let _ =`